demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -865,6 +879,42 @@ static CK_RV prvGetProvisionedState( CK_SESSION_HANDLE xSession,
                                                 &pxProvisionedState->xClientCertificate );
     }
 
+    /* Check for a crypto element identifier. */
+    if( CKR_OK == xResult )
+    {
+        xResult = xGetSlotList( &pxSlotId, &ulSlotCount );
+    }
+
+    if( CKR_OK == xResult )
+    {
+        xResult = pxFunctionList->C_GetTokenInfo( pxSlotId[ 0 ], &xTokenInfo );
+    }
+
+    if( ( CKR_OK == xResult ) && ( '\0' != xTokenInfo.label[ 0 ] ) && ( ' ' != xTokenInfo.label[ 0 ] ) )
+    {
+        /* PKCS #11 requires that token info fields are padded out with space
+         * characters. However, a NULL terminated copy will be more useful to the
+         * caller. */
+        pcSpace = ( char * ) memchr( xTokenInfo.label, ' ', sizeof( xTokenInfo.label ) );
+    }
+
+    if( NULL != pcSpace )
+    {
+        pxProvisionedState->pcIdentifier = ( char * ) pvPortMalloc( 1 + ( pcSpace - ( char * ) xTokenInfo.label ) );
```

Comment:
So... If there is a space, malloc.  If not, do not malloc, but still return CKR_OK potentially?  And what if the label exactly fits the array so there is no space padding?

Reply:
fixed.